src/config/rcxml.c

```diff
@@ -863,8 +869,11 @@ entry(xmlNode *node, char *nodename, char *content)
 		} else {
 			wlr_log(WLR_ERROR, "Invalid value for <resize popupShow />");
 		}
+	} else if (!strcasecmp(nodename, "device.touch")) {
+		touch_config_device_name = xstrdup(content);
 	} else if (!strcasecmp(nodename, "mapToOutput.touch")) {
-		rc.touch.output_name = xstrdup(content);
+		char *output_name = xstrdup(content);
+		touch_config_add(touch_config_device_name, output_name);
```

Comment:
It doesn't feel right to create the object on `mapToOutput.touch`. Technically you could, but if the device-name is given first you have to keep track of the independently which is messy.

Could we handle `touch` elements in `xml_tree_walk()` (i.e. with an `in_touch` state variable and `fill_touch()` function. Suggest searching for + following `in_region` and `fill_region` for an idea of what I'm thinking.

I suspect you'll find it cleaner :smile:

Reply:
yes indeed, much better now, thanks ;)